tower-balance/src/p2c/service.rs

```diff
@@ -307,25 +328,62 @@ where
 }
 
 impl<K, S: Service<Req>, Req> Future for UnreadyService<K, S, Req> {
-    type Item = (K, S);
-    type Error = (K, Error<S::Error>);
+    type Output = Result<(K, S), (K, Error<S::Error>)>;
```

Comment:
Should this now be
```rust
type Output = (K, Result<S, Error<S::Error>>)
```
?

Reply:
I believe we want to own the service in the future then insert it as ready so we want to keep it in the item type.